javascript/javascript_sdk/src/get_data.js

```diff
@@ -0,0 +1,85 @@
+const google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');
+
+const get_zone_data = (client) => {
+    client.listenToObjectUpdate(response => {
+      let zone_lists = response.getZonesList();
+      
+      zone_lists.forEach(zone_list => {
+          let obj = zone_list.getObject();
+          if (zone_list.getType() == 1) {
+              console.log("Enter zone(", zone_list.getId() ,") : ", obj.getId());
+          }
+          else if (zone_list.getType() == 2) {
+              console.log("Exit zone(", zone_list.getId() ,") : ", obj.getId());
+          }
+      });
+    });
+  }
+  
+const get_point_data = (client) => {
+    client.listenToPointCloudUpdate(response => {
+        const pointClouds = response.getPointsList();
+        
+        pointClouds.forEach(pointCloud =>{
+            let pointcloud_info = pointCloud.array;
+            
+            let num_points = Math.floor((pointcloud_info[2].length)/(3.0));
+            if(pointcloud_info[0] == pointMsg.PointResult.PointCloud.Type.RAW){
+            console.log("Topic {%f} no. of points - {%f}", pointcloud_info[1], num_points);
+            }
+            else if(pointcloud_info[0] == pointMsg.PointResult.PointCloud.Type.GROUND){ 
+            console.log("Ground points no. of points - {%f}", num_points);
+            }
+            else if(pointcloud_info[0] == pointMsg.PointResult.PointCloud.Type.ENVIRONMENT){
+            console.log("Environment points no. points - {%f}", num_points);
+            }
+        });
+    });
+}
+  
+const get_object_data = (client) => {
+    client.listenToObjectUpdate(response => {
+      const objects = response.getObjectsList();
+  
+      objects.forEach(object => {
+        let object_info = object.array;
+        if(object_info[100] !== undefined){
```

Comment:
Can you explain this? Isn't this trying to access the 100th element? Or is this code doing something else?

Reply:
yeah I was also bit confused in this part. According to the output of the console it was 6th element of the array, but it was actually accessible only through that particular number.

Reply:
Isn't `object` here a protobuf object? Don't they provide some better way to access the data in the javascript protobuf library?

Reply:
Something like `object.getPoints.length`? I'm not sure about the syntax

Reply:
yeah this works!